src/main/cbindings/CMakeLists.txt

```diff
@@ -0,0 +1,31 @@
+# Reconstruct relevant flags to pass through to LLVM for header parsing.
+# Unfortunately CMAKE_C_FLAGS doesn't include everything; we need to add
+# more, below.
+separate_arguments(LLVM_FLAGS UNIX_COMMAND "${CMAKE_C_FLAGS}")
+
+# Reconstruct "-I" flags.
+get_target_property(INCLUDE_DIRECTORIES shadow-c INCLUDE_DIRECTORIES)
+# Add -I prefix to each directory
+foreach(INCLUDE_DIRECTORY IN LISTS INCLUDE_DIRECTORIES)
+    list(APPEND LLVM_FLAGS "-I${INCLUDE_DIRECTORY}")
+endforeach(INCLUDE_DIRECTORY)
+
+# Reconstruct "-D" flags.
+get_directory_property(COMPILE_DEFINITIONS COMPILE_DEFINITIONS)
+# Add -D prefix to each definition
+foreach(COMPILE_DEFINITION IN LISTS COMPILE_DEFINITIONS)
+    list(APPEND LLVM_FLAGS "-D${COMPILE_DEFINITION}")
```

Comment:
Do we have any headers in Shadow that depend on macros? If so, should we make an issue to remove those?